cloudformation/cw-dashboard-substack.cfn.yaml

```diff
@@ -0,0 +1,554 @@
+            {%- set efs_volume_type = cfn_params.EFSOptions.split(',')[2] %}
+            {%- for metric_condition_params in efs_metrics_conditions_params %}
+              {%- if efs_volume_type in metric_condition_params.supported_vol_types %}
+          - !Sub ',{"type":"metric","x":{{ coord.x }},"y":{{ coord.y }},"width":{{ graph_width }},"height":{{ graph_height }},"properties":{"view":"timeSeries","stacked":false,
+            {{ metric_condition_params.extra_params|join(',') }}{% if metric_condition_params.extra_params|length > 0 %},{% endif %}"region":"${AWS::Region}","metrics":'
+          - !Sub '[["AWS/EFS","{{ metric_condition_params.metric }}","FileSystemId","${EFSFileSystemId}"]]}}'
+              {%- endif %}
+            {%- endfor -%}
+            {{ update_coord(graph_width, graph_height) }}
+
+          {%- endif -%} {#- if efs_shared_dir not in ["NONE", "/NONE"] -#}
+          {{ update_coord_after_section(graph_height) }}
+
+          {%- set fsx_shared_dir = cfn_params.FSXOptions.split(',')[0] %}
+          {%- if fsx_shared_dir not in ["NONE", "/NONE"] %}
+
+          {#- FSx metrics graphs #}
+          - ',{"type":"text","x":{{ coord.x }},"y":{{ coord.y }},"width":24,"height":1,"properties":{"markdown":"\n##
+            FSx Metrics\n"}}'
+            {%- set fsx_metrics = [{'metrics': ["DataReadOperations","DataWriteOperations"], 'extra_params': ['"title":"Data Read/Write Ops"','"period":300']},
+                                   {'metrics': ["DataReadBytes","DataWriteBytes"], 'extra_params': ['"title":"Data Read/Write Bytes"','"period":300']},
+                                   {'metrics': ["FreeDataStorageCapacity"], 'extra_params': ['"title":"Free Data Storage Capacity"']}]
+            -%}
+            {{ update_coord_after_section(1) }}
+            {%- for metrics_param in fsx_metrics %}
+          - !Sub ',{"type":"metric","x":{{ coord.x }},"y":{{ coord.y }},"width":{{ graph_width }},"height":{{ graph_height }},"properties":{"view":"timeSeries","stacked":false,
+            {{ metrics_param.extra_params|join(',') }}{% if metrics_param.extra_params|length > 0 %},{% endif %}"region":"${AWS::Region}","metrics":'
+              {%- for metric in metrics_param.metrics %}
+          - !Sub '{{ '[' if loop.first else ',' }}["AWS/FSx","{{ metric }}","FileSystemId","${FSXFileSystemId}"]{{ ']}}' if loop.last else '' }}'
+              {%- endfor -%}
+              {{- update_coord(graph_width, graph_height) }}
+            {%- endfor %}
+          {%- endif %}
+          {{- update_coord_after_section(graph_height) }}
+
+          {#- Head Node Logs #}
+          - ',{"type":"text","x":{{ coord.x }},"y":{{ coord.y }},"width":24,"height":1,"properties":{"markdown":"\n#
+            Head Node Logs\n"}}'
+          {{- update_coord_after_section(1) }}
+          {%- set dcv_enabled = (cfn_params.DCVOptions.split(',')[0]=='master') %}
```

Comment:
In future instead of using the cfn_params we should use for the conditions the pcluster_config object itself, by retrieving  the dcv section and checking for the enable parameter value:
`pcluster_config.get_section("dcv").get_param_value("enable")`

In this way we'll be decoupled by the CFN parameters representation and instead use the OO representation (Section/Param classes).